src/customer/strategies/apple-pay/apple-pay-customer-strategy.ts

```diff
@@ -0,0 +1,377 @@
+        this._applePayButton.addEventListener('click', this._handleWalletButtonClick);
+
+        return this._store.getState();
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    signIn(): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'In order to sign in via Apple, the shopper must click on "Apple Pay" button.'
+        );
+    }
+
+    signOut(): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'Need to do signout via apple.'
+        );
+    }
+
+    executePaymentMethodCheckout(options?: ExecutePaymentMethodCheckoutOptions): Promise<InternalCheckoutSelectors> {
+        options?.continueWithCheckoutCallback?.();
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _createButton(containerId: string): HTMLElement {
+        const container = document.getElementById(containerId);
+
+        if (!container) {
+            throw new InvalidArgumentError('Unable to create sign-in button without valid container ID.');
+        }
+
+        const button = document.createElement('input');
+        button.type = 'image';
+        button.src = this._paymentMethod?.logoUrl || '';
+        button.style.height = '40px';
+        button.style.width = '75px';
+        container.appendChild(button);
+
+        return button;
+    }
+
+    @bind
+    private _handleWalletButtonClick(event: Event) {
+        event.preventDefault();
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const config = state.config.getStoreConfigOrThrow();
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+        const request = this._getBaseRequest(cart, config, this._paymentMethod);
+        const applePaySession = this._sessionFactory.create(request);
+        this._handleApplePayEvents(applePaySession, this._paymentMethod, config);
+
+        applePaySession.begin();
+    }
+
+    private _getBaseRequest(
+        cart: Cart,
+        config: StoreConfig,
+        paymentMethod: PaymentMethod
+    ): ApplePayJS.ApplePayPaymentRequest {
+        const { storeProfile: { storeCountryCode, storeName } } = config;
+        const { currency: { code } } = cart;
+
+        const { initializationData : { merchantCapabilities, supportedNetworks } } = paymentMethod;
+
+        return {
+            requiredBillingContactFields: ['postalAddress'],
+            requiredShippingContactFields: ['email', 'phone', 'postalAddress'],
+            countryCode: storeCountryCode,
+            currencyCode: code,
+            merchantCapabilities,
+            supportedNetworks,
+            lineItems: [],
+            total: {
+                label: storeName,
+                amount: '0.01',
```

Comment:
Why is the amount hardcoded?

Reply:
This is for initial display of sheet, it's pretty much apple being weird to ask for an amount. This information is not rendered. But we can't pass an empty string as well.

Reply:
And you can't just provide the grand total of the cart?

Reply:
Yes, can do that too 😄